persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/util/MolecularProfileCaseIdentifierUtil.java

```diff
@@ -0,0 +1,33 @@
+package org.cbioportal.persistence.mybatis.util;
+
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Component
+public class MolecularProfileCaseIdentifierUtil {
+
+    public Map<String, Set<String>> getGroupedCasesByMolecularProfileId(List<String> molecularProfileIds, List<String> caseIds) {
+        Map<String, Set<String>> groupMolecularProfileSamples = new HashMap<>();
+
+        if (CollectionUtils.isEmpty(caseIds)) {
+            return molecularProfileIds.stream().collect(Collectors.toMap(Function.identity(), e -> new HashSet<>()));
+        }
+
+        for (int i = 0; i < molecularProfileIds.size(); i++) {
+            String molecularProfileId = molecularProfileIds.get(i);
+            String caseId = caseIds.get(i);
+            if (!groupMolecularProfileSamples.containsKey(molecularProfileId)) {
+                Set<String> filteredCaseIds = new HashSet<>();
+                filteredCaseIds.add(caseId);
+                groupMolecularProfileSamples.put(molecularProfileId, filteredCaseIds);
+            } else {
+                groupMolecularProfileSamples.get(molecularProfileId).add(caseId);
+            }
+        }
+        return groupMolecularProfileSamples;
```

Comment:
Not sure if the stream based solution is simpler, but here it is for your consideration.

```suggestion
        return IntStream.range(0, molecularProfileIds.size())
            .mapToObj(i -> new Pair<>(molecularProfileIds.get(i), caseIds.get(i)))
            .collect(Collectors.groupingBy(
                Pair<String, String>::getA,
                Collectors.mapping(Pair<String, String>::getB, toSet()))
            );
```